pkg/controller/infrastructure/infraflow/delete.go

```diff
@@ -96,48 +103,47 @@ func (fctx *FlowContext) buildDeleteGraph() *flow.Graph {
 }
 
 func (fctx *FlowContext) deleteRouter(ctx context.Context) error {
-	log := shared.LogFromContext(ctx)
-	current, err := fctx.findExistingRouter()
-	if err != nil {
-		return err
+	routerID := fctx.state.Get(IdentifierRouter)
+	if routerID == nil {
+		return nil
 	}
-	if current != nil {
-		log.Info("deleting...", "router", current.ID)
-		if err := fctx.networking.DeleteRouter(current.ID); err != nil {
-			return err
-		}
+
+	shared.LogFromContext(ctx).Info("deleting...", "router", *routerID)
+	if err := fctx.networking.DeleteRouter(*routerID); err != nil {
```

Comment:
This refactoring leads to a new corner case if the router is deleted but the routerID is still contained in the state.

In the old version, `findExistingRouter` would return `nil` and thus the router deletion is just skipped. With the new variant, if the routerID is still part of the state (e.g. router deletion completed, but the call ran into a timeout and thus didn't clear the state) then then `DeleteRouter` call would run into an error. In my opinion, the error should be filtered with `client.IgnoreNotFoundError(err)`.

Reply:
Will look into it. Thank you @MichaelEischer